src/cata_imgui.cpp

```diff
@@ -461,6 +462,13 @@ class cataimgui::window_impl
         }
 };
 
+class cataimgui::filter_box_impl
+{
+    public:
+        char text[255]; // NOLINT(modernize-avoid-c-arrays)
```

Comment:
I mean, c'mon.
```suggestion
        std::array<char, 255> text;
```

Reply:
the c-style array here came from input_context.cpp where I have vivid memories of trying to do this in the original ImGui PR and it not working and having to fall back on this. But I will give it another try

Reply:
done, not sure why we nuked that before